xtask/src/custom_workload.rs

```diff
@@ -34,9 +37,27 @@ pub fn new_custom_workload(
         })
         .collect();
 
-    let read_actions = (0..n_reads).map(|i| Action::Read { key: rand_key(i) });
+    let rand_or_seq = |i: u64, prev: &mut Option<Uint<256, 4>>| -> Vec<u8> {
```

Comment:
for example, what if you created the RNG here and moved it into the closure, passing it to `rand_key` each time?
it's OK if it is never used in the sequential branch.